docs/ref/models/querysets.txt

```diff
@@ -2026,10 +2026,28 @@ won't cache results after iterating over them. Oracle and :ref:`PostgreSQL
 <postgresql-server-side-cursors>` use server-side cursors to stream results
 from the database without loading the entire result set into memory.
 
+``chunk_size`` specifies the number of results to cache at the database driver
+level. Fetching bigger chunks diminishes the number of round trips between the
+database driver and the database, at the expense of memory. For database
+backends that don't support server-side cursors, this argument is used to
+control size of batches retrieved from the database driver.
+
+The default value of 2000 comes from `this calculation`_:
```

Comment:
I'd say something like "a calculation on the psycopg mailing list" to make it a bit more clear where the link is headed.